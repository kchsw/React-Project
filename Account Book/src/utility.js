export const LIST_VIEW = 'list'
export const CHART_VIEW = 'chart'
export const TYPE_INCOME = 'income'
export const TYPE_OUTCOME = 'outcome'
export const padLeft = (n) => {
    return n < 10 ? '0' + n : n
}

export const range = (size, startAt = 0) => {
    const arr = []
    for(let i = 0; i < size; i++){
        arr[i] = startAt + i
    }
    return arr
}

export const parseDate = (str) => {
    const date = str? new Date(str) : new Date()
    return {
        year: date.getFullYear(),
        month: date.getMonth() + 1
    }

}

export const Colors = {
    blue: '#347eff',
    deepBlue: '#61dafb',
    green: '#28a745',
    red: '#dc3545',
    gray: '#555',
    lightGray: '#efefef',
    white: '#fff'
}