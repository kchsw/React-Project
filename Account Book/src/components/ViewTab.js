import React from 'react'
import Ionicon from 'react-ionicons'
import PropTypes from 'prop-types'
import { LIST_VIEW, CHART_VIEW } from '../utility'

const generateLinkClass = (current, view) => {
    return current === view? "nav-link active" : "nav-link"
}



const ViewTab = ({ activeTab, onTabChange }) => {
    const tabClick = (event, view) => {
        event.preventDefault();
        onTabChange(view)
    }
    return (
        <ul className="nav nav-tabs nav-fill my-4">
            <li className="nav-item list-item">
                <a className={generateLinkClass(activeTab, LIST_VIEW)} 
                    href="#"
                    onClick={(e) => {e.preventDefault(); onTabChange(LIST_VIEW)}}
                >
                    <Ionicon
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={'#007bff'}
                        icon="ios-paper"
                    />
                    列表模式
                </a>
            </li>
            <li className="nav-item chart-item">
                <a className={generateLinkClass(activeTab, CHART_VIEW)} 
                    href="#"
                    onClick={(e) => {e.preventDefault(); onTabChange(CHART_VIEW)}}
                    // onClick={(e) => {tabClick(e, CHART_VIEW)}}
                >
                    <Ionicon
                        className="rounded-circle mr-2"
                        fontSize="25px"
                        color={'#007bff'}
                        icon="ios-pie"
                    />
                   图表模式
                </a>
            </li>
        </ul>
    )
}

ViewTab.propTypes = {
    activeTab: PropTypes.string.isRequired,
    onTabChange: PropTypes.func.isRequired,
}

export default ViewTab