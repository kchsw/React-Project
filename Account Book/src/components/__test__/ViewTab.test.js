import React from 'react';
import { shallow } from 'enzyme';
import ViewTab from '../ViewTab'
import { LIST_VIEW, CHART_VIEW } from '../../utility'

const props = {
    activeTab: LIST_VIEW, 
    onTabChange: jest.fn()
}

let wrapper 

describe('test ViewTab component', () => {
    beforeEach(() => {
        wrapper = shallow(<ViewTab {...props}/>)
    })
    it('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('render the correct actived tab', () => {
       expect(wrapper.find('.list-item .nav-link').first().hasClass('active')).toBeTruthy()
       expect(wrapper.find('.chart-item').length).toEqual(0)
    })
    it('should trigger the correct function callbacks', () => {
        wrapper.find('.chart-item .nav-link').first().simulate('click')
        expect(props.onTabChange).toHaveBeenCalledWith(CHART_VIEW)
        expect(wrapper.state('activeTab')).toEqual('CHART_VIEW')
    })
})
