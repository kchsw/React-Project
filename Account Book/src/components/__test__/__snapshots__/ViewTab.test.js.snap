// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ViewTab component snapshot 1`] = `ShallowWrapper {}`;
